### ISSUE TITLE
Error when creating a summary inside a tf.while_loop in a multi-GPU setup (like CIFAR-10)

### ISSUE THREAD

**Issue by reuben** [tensorflow/tensorflow#8660]
Why diagnosing a problem with batch normalized RNNs, I tried to add a summary of the population statistics in my batch normalized cell, but then when I try to run the merged summaries I get the following error:

tensorflow.python.framework.errors_impl.InvalidArgumentError: The node 'Merge/MergeSummary' has inputs from different frames. The input 'tower_2/rnn/while/multi_rnn_cell/cell_4/gru_cell/gates/r/rnn/multi_rnn_cell/cell_4/gru_cell/gates/r/pop_var_0' is in frame 'tower_2/rnn/while/tower_2/rnn/while/'. The input 'tower_3/rnn/while/multi_rnn_cell/cell_4/gru_cell/gates/r/rnn/multi_rnn_cell/cell_4/gru_cell/gates/r/pop_var_0' is in frame 'tower_3/rnn/while/tower_3/rnn/while/'.

This is on TensorFlow 1.0. I'll try to get a reduced testcase.

**Comment by reuben** [comment]
This only happens when I have multiple towers. Masking GPUs with CUDA_VISIBLE_DEVICES so that only tower_0 is created makes the problem go away.

**Comment by aselle** [comment]
@dandelionmane, is it expected that summaries cannot be created in while loops?

**Comment by userisanumber** [comment]
I have the same problem when I use `summary.merge` for variables which were created with `while_loop `

tensorflow.python.framework.errors_impl.InvalidArgumentError: The node 'Merge/MergeSummary' has inputs from different frames. The input 'tower_0/while/o_rnn_3/kern/HistogramSummary_1' is in frame 'tower_0/while/tower_0/while/'. The input 'g_embatch_size/emb_x_0' is in frame ''

**Comment by reuben** [comment]
@itsmeolivia, why was this closed? It looks like a similar but different problem than #10059.

**Comment by itsmeolivia** [comment]
oops, the tags were wrong. reopening and re-tagging

**Comment by dandelionmane** [comment]
Yes, this is a known issue that conditional ops don't play well with summary ops. 
@alextp @josh11b @ali01 

I'm removing the `tensorboard` tag because it's not a tensorboard issue, it's a problem with core tensorflow.

**Comment by cuter9** [comment]
when I used summary.merge(loss) to generate a summary string, it raise the type error 'Tensor' object is not iterable. ' It seems while loop and any iterator are not supported by the merge method? is there any solutions, or alternatives?

**Comment by ppwwyyxx** [comment]
@cuter9 The [document](https://www.tensorflow.org/api_docs/python/tf/summary/merge) says: inputs: A list of __string Tensor__ objects containing serialized __Summary protocol buffers__.

**Comment by cuter9** [comment]
The following is my scrips of a function which is called iteratively thru for loop, still don't work, and raise type error " 'Tensor' object is not iterable", and the dubuger points to the problem raising from tf.summary.merge(x_summary) statement.

def summary_4_graph(data_x):
    x_summary = tf.summary.scalar(data_x.name, data_x)
    tf.summary.merge(x_summary)

**Comment by ppwwyyxx** [comment]
A __list__ of string Tensor objects containing serialized Summary protocol buffers.

**Comment by cuter9** [comment]
done! thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activityand the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.
